src/assets/assets.service.ts

```diff
@@ -20,7 +23,19 @@ export class AssetsService {
     return record;
   }
 
-  async findByIdentifierOrThrow(
+  async findByIdentifierOrThrow(identifier: string): Promise<Asset> {
+    const record = await this.prisma.asset.findUnique({
+      where: {
+        identifier,
+      },
+    });
+    if (!record) {
+      throw new NotFoundException();
+    }
+    return record;
+  }
+
+  async findByIdentifierOrThrowWithClient(
```

Comment:
Can we merge these two methods and use a default instead of having two?

Reply:
I'd like to keep as is since we follow this pattern across other services in our API and the method name is explicit to indicate when a transaction client might be needed. I'm open to changing it but would probably want to have that discussion as a separate change - what do you think?

Reply:
Fine with me, I thought I remember seeing the opposite pattern in the code elsewhere, but if this is the standard then yea lets keep it this way